File: examples/2015-day-01/ocaml/test/test_aoc.ml
open Alcotest

let test_part_one expected lines =
  fun () -> check int "solves part one" expected (Aoc.solve_part_one lines)
;;

let test_part_two expected lines =
  fun () -> check int "solves part two" expected (Aoc.solve_part_two lines)
;;

let () =
  run
    "AoC"
    [ ( "Part One"
      , [ test_case "Example 1" `Quick (test_part_one 0 [ "(())" ])
        ; test_case "Example 2" `Quick (test_part_one 0 [ "()()" ])
        ; test_case "Example 3" `Quick (test_part_one 3 [ "(((" ])
        ; test_case "Example 4" `Quick (test_part_one 3 [ "(()(()(" ])
        ; test_case "Example 5" `Quick (test_part_one 3 [ "))(((((" ])
        ; test_case "Example 6" `Quick (test_part_one (-1) [ "())" ])
        ; test_case "Example 7" `Quick (test_part_one (-1) [ "))(" ])
        ; test_case "Example 8" `Quick (test_part_one (-3) [ ")))" ])
        ; test_case "Example 9" `Quick (test_part_one (-3) [ ")())())" ])
        ] )
    ; ( "Part Two"
      , [ test_case "Example 1" `Quick (test_part_two 1 [ ")" ])
        ; test_case "Example 2" `Quick (test_part_two 5 [ "()())" ])
        ] )
    ]
;;

File: examples/2015-day-01/shakespeare/tests/test_aoc.py
import pytest

from aoc import solve_part_one, solve_part_two

part_one_examples = [
    ("(())", 0),
    ("()()", 0),
    ("(((", 3),
    ("(()(()(", 3),
    ("))(((((", 3),
    ("())", -1),
    ("))(", -1),
    (")))", -3),
    (")())())", -3),
]
part_two_examples = [
    (")", 1),
    ("()())", 5),
]


@pytest.mark.parametrize(("puzzle_input", "expected"), part_one_examples)
def test_part_one(puzzle_input: str, expected: int) -> None:
    assert solve_part_one(puzzle_input) == expected


@pytest.mark.parametrize(("puzzle_input", "expected"), part_two_examples)
def test_part_two(puzzle_input: str, expected: int) -> None:
    assert solve_part_two(puzzle_input) == expected
